src/middleware/r2Middleware.ts

```diff
@@ -107,9 +108,22 @@ async function getFile(
   r2Path: string,
   ctx: Context
 ): Promise<Response> {
-  const result = await getProvider(ctx).getFile(r2Path, {
-    conditionalHeaders: parseConditionalHeaders(request.headers),
-  });
+  const provider = getProvider(ctx);
+
+  let result: GetFileResult | undefined;
+  try {
```

Comment:
Instead of try catch can we use promises here?

Reply:
Because it would look cleaner keeping the same code with a .catch() and handling the scenarios without all this extra code

Reply:
Since this returns a response in the catch scope we can't